keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -777,127 +788,147 @@ kind: FlowSchema
+      name: "system:nodes"    
+  - rule:
+      verbs: ['*']
+      apiGroups: ['coordination.k8s.io']
+      resources: ['leases']
+    subjects:
+    - kind: User
+      name: "system:kube-controller-manager"    
+    - kind: User
+      name: "system:kube-scheduler" 
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
   name: system-low
 spec:
-  matchingPriority: 900
-  requestPriority:
+  matchingPrecedence: 3500
+  priorityLevelConfiguration:
     name: system-low
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and: # the garbage collector
-    - equals:
-      field: user
-      value: system:controller:garbage-collector
+  distinguisherMethod:
+    type: ByUser
+  rules:  
+  - rule:
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: User
+      name: "system:kube-controller-manager"    
+    - kind: User
+      name: "system:kube-scheduler"    
+  - rule:
+      verbs: ['*']
+      nonResourceURLs: ['*']
+    subjects:
+    - kind: User
+      name: "system:kube-controller-manager"    
+    - kind: User
+      name: "system:kube-scheduler"         
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
-  name: workload-high
+  name: workload
 spec:
-  requestPriority:
-    name: workload-high
-  flowDistinguisher:
-    source: namespace
-    # no transformation in this case
-  match:
-  - and: # users using kubectl
-    - notPatternMatch:
-      field: user
-      pattern: system:serviceaccount:.*
+  matchingPrecedence: 7500
+  priorityLevelConfiguration:
+    name: workload
+  distinguisherMethod:
+    type: ByNamespace
+  rules:  
+  - rule:
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:serviceaccounts"
+  - rule:
+      verbs: ['*']
+      nonResourceURLs: ['*']
+    subjects:
+    - kind: Group
+      name: "system:serviceaccounts"      
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
-  name: workload-low
-spec:
-  matchingPriority: 9999
-  requestPriority:
-    name: workload-high
-  flowDistinguisher:
-    source: namespace
-    # no transformation in this case
-  match:
-  - and: [ ] # match everything
-```
-  
-Following is a FlowSchema that might be used for the requests by the
-aggregated apiservers of
-https://github.com/MikeSpreitzer/kube-examples/tree/add-kos/staging/kos
-to create TokenReview and SubjectAccessReview objects.
-
-
-```
-kind: FlowSchema
-meta:
-  name: system-top
+  name: catch-all
 spec:
-  matchingPriority: 900
-  requestPriority:
-    name: system-top
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and:
-    - inSet:
-      field: resource
-      set: [ "tokenreviews", "subjectaccessreviews" ]
-    - superSet:
-      field: user
-      set: [ "system:serviceaccount:example-com:network-apiserver" ]
+  matchingPrecedence: 10000
+  priorityLevelConfiguration:
+    name: default
+  distinguisherMethod:
+    type: ByUser
```

Comment:
In the original this was by object namespace rather than username.  Why the change?

Reply:
oversight. fixed